src/mono/mono/mini/debugger-agent.c

```diff
@@ -9417,6 +9004,13 @@ array_commands (int command, guint8 *p, guint8 *end, Buffer *buf)
 		return err;
 
 	switch (command) {
+	case CMD_ARRAY_REF_GET_TYPE: {
+			buffer_add_byte(buf, m_class_get_byval_arg (m_class_get_element_class (arr->obj.vtable->klass))->type);
+			buffer_add_int (buf, m_class_get_rank (arr->obj.vtable->klass));
+			if (m_class_get_byval_arg (m_class_get_element_class (arr->obj.vtable->klass))->type == MONO_TYPE_CLASS)
```

Comment:
what if it's an array of generic instances?

Reply:
I didn't think about it yet, do you think I need to think right now?

Reply:
> I didn't think about it yet, do you think I need to think right now?

I'm not sure. What's the failure mode if you get something unexpected? will the other side be able to do something reasonable without the type? If the debugger won't crash, then it's probably ok to do later.